server/routes/replayEventRouter.js

```diff
@@ -0,0 +1,42 @@
+const express = require('express');
+const replayEventRouter = express.Router();
+
+const { getDomainEvents, dispatch } = require('../models/domain-event');
+
+const Session = require('../infra/database/Session');
+const { publishMessage } = require('../infra/messaging/RabbitMQMessaging');
+
+const { EventRepository } = require('../infra/database/PgRepositories');
+const { dictEventHandlers, handleEvent} = require('../services/event-handlers');
+
+const processEvent = (event, executor) => {
+    try {
+        executor(event);
+    } catch (e) {
+        console.log(`Error processing event ${e}`);
+    }
+}
+
+replayEventRouter.post("/", async function(req, res) {
+    const session = new Session(false);
+    const eventRepository = new EventRepository(session);
+    const executeGetDomainEvents = getDomainEvents(eventRepository);
+    const domainEvents = await executeGetDomainEvents({"status":req.body.status});
+    if(req.body.status == "raised"){
+        const eventDispatch = dispatch(eventRepository, publishMessage);
+            domainEvents.forEach(async (domainEvent) => {
+                processEvent(domainEvent.payload,eventDispatch);
+            });
+    }else{
+        domainEvents.forEach(async (domainEvent) => {

```

Comment:
Based on the changes I am recommending in event-handler.js, the else block can instead be using the code below. The change here is to use domainEvent as the parameter to the event handler instead of `domainEvent.payload`.
The other change is to not use local function `processEvent` since I don't see the utility of that. Unless if I am missing something.
 
```
  domainEvents.forEach(async (domainEvent) => {
            const eventHandler = dictEventHandlers[domainEvent.payload.type];
            if (typeof eventHandler !== "undefined" && eventHandler !== null){
                const executeHandleEvent = handleEvent(eventHandler);
                executeHandleEvent(domainEvent);
            }
        });
```